tests/proxy.rs

```diff
@@ -45,15 +46,40 @@ async fn test_shutdown_lifecycle() {
         .expect("app exits without error")
 }
 
-#[tokio::test]
-async fn test_conflicting_bind_error() {
+// Check that port conflicts on any address results in the app failing instead of silently failing
+async fn test_bind_conflict<F: FnOnce(&mut ztunnel::config::Config) -> &mut SocketAddr>(f: F) {
```

Comment:
Would make it more fine granularity by splitting into two functions, first allocate a idle port and second set to Config